README.md

```diff
@@ -1,158 +1,339 @@
 
-> **Note**: The provided handler implementation is only meant to demonstrate the use of the SDK APIs and is not representative of the best practices of a reconciliation loop.
+### Resources watched by the Controller
 
-### Build and run the Operator
+Inspect the Controller implementation at `pkg/controller/memcached/memcached_controller.go` to see how the Controller watches resources.
 
-Build the `memcached-operator` image and push it to a registry. Please make sure you have a an account on [Quay.io][quay_io] for the next step, or substitute your preferred container registry. On the registry, [create a new public image][create_public_image] repository named “memcached-operator”.
+The first watch is for the `Memcached` type as the primary resource. For each Add/Update/Delete event the reconcile loop will be sent a reconcile `Request` (a namespace/name key) for that `Memcached` object:
+
+```Go
+err := c.Watch(
+    &source.Kind{Type: &cachev1alpha1.Memcached{}},
+    &handler.EnqueueRequestForObject{},
+  )
+```
 
+The next watch is for Deployments but the event handler will map each event to a reconcile `Request` for the owner of the Deployment. Which in this case is the `Memcached` object for which the Deployment was created. This allows the controller to watch Deployments as a secondary resource.
+
+```Go
+err := c.Watch(
+    &source.Kind{Type: &appsv1.Deployment{}},
+    &handler.EnqueueRequestForOwner{
+        IsController: true,
+        OwnerType:    &cachev1alpha1.Memcached{}},
+    )
+```
+
+### Reconcile loop
+
+Every Controller has a Reconciler object with a `Reconcile()` method that implements the reconcile loop. The reconcile loop is passed the [`Request`][request_go_doc] argument which is a Namespace/Name key used to lookup the primary resource object, `Memcached`, from the cache:
+
+```Go
+func (r *ReconcileMemcached) Reconcile(request reconcile.Request) (reconcile.Result, error) {
+    // Lookup the Memcached instance for this reconcile request
+    memcached := &cachev1alpha1.Memcached{}
+    err := r.client.Get(context.TODO(), request.NamespacedName, memcached)
+    ...
+}  
+```
+
+For a guide on Reconcilers, Clients, and interacting with resource Events, see the [Client API doc][doc_client_api].
+
+## Build and run the operator
+
+Before running the operator, the CRD must be registered with the Kubernetes apiserver:
+
+```sh
+$ kubectl create -f deploy/crds/cache_v1alpha1_memcached_crd.yaml
+```
+
+Once this is done, there are two ways to run the operator:
+
+* As a Deployment inside a Kubernetes cluster
+* As Go program outside a cluster
+
+### 1. Run as a Deployment inside the cluster
+
+Build the memcached-operator image and push it to a registry:
 
 ```sh
-$ cd $GOPATH/src/github.com/example-inc/
-$ cd memcached-operator
 $ operator-sdk build quay.io/example/memcached-operator:v0.0.1
+$ sed -i 's|REPLACE_IMAGE|quay.io/example/memcached-operator:v0.0.1|g' deploy/operator.yaml
```

Comment:
Maybe add the note about OSX? https://github.com/operator-framework/operator-sdk/pull/699